### PR TITLE
Manage mxnet context when deserializing predictors

### PR DESCRIPTION
**Pull request by vafl**
*Issue #, if available:*
#313 

*Description of changes:*
Manage mxnet context when deserializing predictors and don't swallow error messages.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-318/1 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-318/1/index.html

File: src/gluonts/model/predictor.py [hunk 1]
```diff
@@ -106,25 +106,28 @@ def predict(self, dataset: Dataset, **kwargs) -> Iterator[Forecast]:
         raise NotImplementedError
 
     def serialize(self, path: Path) -> None:
-        try:
-            # serialize Predictor type
-            with (path / "type.txt").open("w") as fp:
-                fp.write(fqname_for(self.__class__))
-        except Exception as e:
```
Comment (lostella) [inline src/gluonts/model/predictor.py]: Why do all the try-catch need to be removed?

Comment (vafl) [reply src/gluonts/model/predictor.py]: They are swallowing actual error messages and stack traces and replace them by some generic error message.

Comment (vafl) [reply src/gluonts/model/predictor.py]: Maybe we can add a test, since the build also runs on a GPU instance...

**Pull request merged**